File: WindowsRegular Expressions/4975sourcecode/chapter4/4-10/recipe4-10.vb
Imports System
Imports System.IO
Imports System.Text.RegularExpressions
Public Class Recipe

	Private Const octetRegex = "([1-9]?[0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])"
    Private Shared _Regex As Regex = _
    	New Regex(String.Format("^{0}\.{0}\.{0}\.{0}$", octetRegex))

    Public Sub Run(ByVal fileName As String)
        Dim line As String
        Dim lineNbr As Integer = 0
        Dim sr As StreamReader = File.OpenText(fileName)
        line = sr.ReadLine
        While Not line Is Nothing
            lineNbr = lineNbr + 1
            If _Regex.IsMatch(line) Then
                Console.WriteLine("Found match '{0}' at line {1}", _
                    line, _
                    lineNbr)
            End If
            line = sr.ReadLine
        End While
        sr.Close()
    End Sub

    Public Shared Sub Main(ByVal args As String())
        Dim r As Recipe = New Recipe
        r.Run(args(0))
    End Sub

End Class

File: WindowsRegular Expressions/4975sourcecode/chapter4/4-10/recipe4-10.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

public class Recipe
{
	private static string octetRegex = "([1-9]?[0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])";
	private static Regex _Regex = 
		new Regex(String.Format(@"^{0}\.{0}\.{0}\.{0}$", octetRegex));
	
	public void Run(string fileName)
	{
		String line;
		int lineNbr = 0;
		using (StreamReader sr = new StreamReader(fileName))
		{
			while(null != (line = sr.ReadLine()))
			{
				lineNbr++;
				if (_Regex.IsMatch(line))
				{
					Console.WriteLine("Found match '{0}' at line {1}", 
						line, 
						lineNbr);
				}
			}
		}
	}	

	public static void Main( string[] args )
	{
		Recipe r = new Recipe();
		r.Run(args[0]);
	}
}
